src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5821,13 +5821,23 @@ static QVariant fcnFormatDate( const QVariantList &values, const QgsExpressionCo
 
 static QVariant fcnColorGrayscaleAverage( const QVariantList &values, const QgsExpressionContext *, QgsExpression *, const QgsExpressionNodeFunction * )
 {
-  QColor color = QgsSymbolLayerUtils::decodeColor( values.at( 0 ).toString() );
-  int avg = ( color.red() + color.green() + color.blue() ) / 3;
-  int alpha = color.alpha();
+  const QVariant variant = values.at( 0 );
```

Comment:
This logic should be moved to QgsExpressionUtils, as getColorValue, with isQColor as a by reference argument

Reply:
Done 06048538d4d